### PR TITLE
GitHub Actions for CanDIGv2

### PR DESCRIPTION
**Pull request by justin-ys**
I wanted to finish implementing workflow dispatches before this PR but unfortunately I need to commit the action to the main branch at least once first; I think it should be finished regardless but unfortunately I may need to make another PR if it requires fixes.

This PR creates a GitHub action which builds the CanDIGv2 stack, ensures all containers are running, and runs integration tests. It is triggered on PRs/commits to main as well as (if working) commits to the main repositories of submodules. Build and container logs are saved as an artifact within the repo.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,9 +15,8 @@ SHELL = bash
 #  and then use make mkdir and make init-conda (no bin-conda, which will blow up an existing conda)
 # <<<
 
-CONDA_INSTALL = bin
```
Comment (daisieh) [inline Makefile:18]: Is removing this line going to be a problem for other platforms?

Comment (daisieh) [reply Makefile:18]: Oh I see, it moved to .env

File: etc/tests/test_integration.py [hunk 2]
```diff
@@ -163,6 +163,7 @@ def test_vault():
 
     # confirm that this works with the CANDIG_S3_TOKEN too:
     headers["X-Vault-Token"] = ENV["VAULT_S3_TOKEN"]
+    print(f"vault token is {headers['X-Vault-Token']}")
```
Comment (daisieh) [inline etc/tests/test_integration.py]: This can probably be removed...

**Review by daisieh — CHANGES_REQUESTED** [review]
Can you do a review of https://github.com/CanDIG/CanDIGv2/pull/247 so that we can merge that one first? Then you can rebase this branch on develop and we can keep this PR clean.

File: etc/tests/test_integration.py (hunk 2 above)
Comment (justin-ys) [reply etc/tests/test_integration.py]: Resolved by [639c87b](https://github.com/CanDIG/CanDIGv2/pull/248/commits/639c87b8b1255f57ab21b2fa1b28c05d70b20a50)

**Review by daisieh — APPROVED** [review]
Looks good!

**Pull request merged**